W-14795609 Connector ACB Configuration (#2478)
mulesoft/docs-connectors: connectors-home/modules/ROOT/partials/acb-config-db-general-tab.adoc

// Used in database-connector-acb for the common properties in the General tab

// tag::general-tab-attributes[]
. In the *General* tab, configure the following attributes:
+
[%header%autowidth.spread]
|===
| UI Field | Description
| Input Parameters | A map in which keys are the name of an input parameter to set on the JDBC prepared statement. Each parameter must be referenced in the SQL text using a colon prefix, for example, `where id = :myParamName`. The map's values contain the actual assignation for each parameter.
| SQL Query Text |Text of the SQL query to execute.
|===

// end::general-tab-attributes[]

// tag::parameter-type[]
[[ParameterType]]
=== Parameter Type

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Field | Type | Description | Default Value | Required
| Key a| String | The name of the input parameter |  | x
| Type Classifier a| <<TypeClassifier>> |  |  | x
|===
// end:parameter-type[]

// tag::advanced-tab[]
. Select the *Advanced* tab to configure the following attributes:
+
[%header%autowidth.spread]
|===
| UI Field | Description 

> | Transactional action | The type of joining action that operations can take for transactions:

Reviewer: ```suggestion
| Transactional action a| The type of joining action that operations can take for transactions:
```

> 

Reviewer: ```suggestion
+
```

> +

Reviewer: ```suggestion

```

** ALWAYS_JOIN
** JOIN_IF_POSSIBLE (Default)
** NOT_SUPPORTED
| Query timeout | Indicates the minimum amount of time before the JDBC driver attempts to cancel a running statement. By default, no timeout (`0`) is used.
| Query timeout unit | Select the unit of time to use. The default is SECONDS.
| Fetch size | Indicates how many rows to fetch from the database when rows are read from a `ResultSet`. This property is required when streaming is set to `true`, in which case a default value of `10` is used.
|Max rows | Sets the limit for the maximum number of rows that any `ResultSet` object generated by the message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped.

> |Parameter types | Array of xref:database-documentation.adoc#ParameterType[ParameterType]. Use this parameter to optionally specify the type of (one or more) of the parameters in the query. If a value is provided, you're not required to reference all of the parameters. You can't reference a parameter that is not present in the input values. 

Reviewer: ```suggestion
|Parameter types | Array of xref:database-documentation.adoc#ParameterType[Parameter Type]. Use this parameter to optionally specify the type of (one or more) of the parameters in the query. If a value is provided, you're not required to reference all of the parameters. You can't reference a parameter that is not present in the input values. 
```

| Target Variable | Name of the variable that stores the operation's output. |  
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. 
| Error Mappings | This attribute is not currently enabled. 

> |Reconnection Strategy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:database-documentation.adoc#reconnect[Reconnect] and xref:database-documentation.adoc#reconnect[Reconnect Forever]. 

Reviewer: ```suggestion
|Reconnection Strategy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:database-documentation.adoc#reconnect[Reconnect], xref:database-documentation.adoc#reconnect[Reconnect Forever], and <<configuring-reconnection-strategy>>.
```

|===

// end::advanced-tab[]

// tag::advanced-tab-streaming-strategy[]
. Select the *Advanced* tab to configure the following attributes:
+
[%header%autowidth.spread]
|===
| UI Field | Description 

> |Streaming Strategy | This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:connectors::streaming-strategies.adoc[].

Reviewer: ```suggestion
|Streaming Strategy | This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See include::connectors::partial$streaming-strategies.adoc[Streaming Strategies].
```

> | Transactional action | The type of joining action that operations can take for transactions:

Reviewer: ```suggestion
| Transactional action a| The type of joining action that operations can take for transactions:
```

> 

Reviewer: ```suggestion
+
```

> +

Reviewer: ```suggestion

```

** ALWAYS_JOIN
** JOIN_IF_POSSIBLE (Default)
** NOT_SUPPORTED
| Query timeout | Indicates the minimum amount of time before the JDBC driver attempts to cancel a running statement. By default, no timeout (`0`) is used.
| Query timeout unit | Select the unit of time to use. The default is SECONDS.
| Fetch size | Indicates how many rows to fetch from the database when rows are read from a `ResultSet`. This property is required when streaming is set to `true`, in which case a default value of `10` is used.
|Max rows | Sets the limit for the maximum number of rows that any `ResultSet` object generated by the message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped.

> |Parameter types | Array of xref:database-documentation.adoc#ParameterType[ParameterType]. Use this parameter to optionally specify the type of (one or more) of the parameters in the query. If a value is provided, you're not required to reference all of the parameters. You can't reference a parameter that is not present in the input values. 

Reviewer: ```suggestion
|Parameter types | Array of xref:database-documentation.adoc#ParameterType[Parameter Type]. Use this parameter to optionally specify the type of (one or more) of the parameters in the query. If a value is provided, you're not required to reference all of the parameters. You can't reference a parameter that is not present in the input values. 
```

| Target Variable | Name of the variable that stores the operation's output. |  
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. 
| Error Mappings | This attribute is not currently enabled. 
|Reconnection Strategy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:database-documentation.adoc#reconnect[Reconnect] and xref:database-documentation.adoc#reconnect[Reconnect Forever] and <<configuring-reconnection-strategy>> 
|===
// end::advanced-tab-streaming-strategy[]

// tag::advanced-tab-autogenerate-keys[]
. Select the *Advanced* tab to configure the following attributes:
+
[%header%autowidth.spread]
|===
| UI Field | Description 

> | Transactional action | The type of joining action that operations can take for transactions:

Reviewer: ```suggestion
| Transactional action a| The type of joining action that operations can take for transactions:
```

> 

Reviewer: ```suggestion
+
```

> +

Reviewer: ```suggestion

```

** ALWAYS_JOIN
** JOIN_IF_POSSIBLE (Default)
** NOT_SUPPORTED
| Query timeout | Indicates the minimum amount of time before the JDBC driver attempts to cancel a running statement. By default, no timeout (`0`) is used.
| Query timeout unit | Select the unit of time to use. The default is SECONDS.
| Fetch size | Indicates how many rows to fetch from the database when rows are read from a `ResultSet`. This property is required when streaming is set to `true`, in which case a default value of `10` is used.
|Max rows | Sets the limit for the maximum number of rows that any `ResultSet` object generated by the message processor can contain for the given number. If the limit is exceeded, the excess rows are silently dropped.

> |Parameter types | Array of xref:database-documentation.adoc#ParameterType[ParameterType]. Use this parameter to optionally specify the type of (one or more) of the parameters in the query. If a value is provided, you're not required to reference all of the parameters. You can't reference a parameter that is not present in the input values. 

Reviewer: ```suggestion
|Parameter types | Array of xref:database-documentation.adoc#ParameterType[Parameter Type]. Use this parameter to optionally specify the type of (one or more) of the parameters in the query. If a value is provided, you're not required to reference all of the parameters. You can't reference a parameter that is not present in the input values. 
```

| Auto Generate Keys |Indicates when to make auto-generated keys available for retrieval. 
| Auto Generated Keys Column Indexes | List of column indexes that indicates which auto-generated keys to make available for retrieval. This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:database-documentation.adoc#insert[Insert].
| Auto Generated Keys Column Names | List of column names that indicates which auto-generated keys to make available for retrieval. This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:database-documentation.adoc#insert[Insert].
| Target Variable | Name of the variable that stores the operation's output. |  
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. 
| Error Mappings | This attribute is not currently enabled. 

> |Reconnection Strategy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:database-documentation.adoc#reconnect[Reconnect] and xref:database-documentation.adoc#reconnect[Reconnect Forever]. 

Reviewer: ```suggestion
|Reconnection Strategy |This attribute is not currently enabled in the configuration UI. Configure it in the XML editor. See xref:database-documentation.adoc#reconnect[Reconnect], xref:database-documentation.adoc#reconnect[Reconnect Forever], and <<configuring-reconnection-strategy>>.
```
